packages/transformers/assemblyscript/src/ASTransformer.js

```diff
@@ -0,0 +1,144 @@
+// @flow strict-local
+
+import {Transformer} from '@parcel/plugin';
+import ThrowableDiagnostic from '@parcel/diagnostic';
+import {promises as fsp} from 'fs';
+import path from 'path';
+import {tmpdir} from 'os';
+import {Transform} from 'stream';
+
+const MARKER = 'asc:';
+const PREFIX_MATCHER = /^asc:(.+)$/;
+
+// This special import contains the `compileStreaming` polyfill.
+const SPECIAL_IMPORT = '__rollup-plugin-assemblyscript_compileStreaming';
+
+const compileStreamingCode = `
+  export async function compileStreaming(respP) {
+              if('compileStreaming' in WebAssembly) {
+                return WebAssembly.compileStreaming(respP);
+              }
+              return respP
+                .then(resp => resp.arrayBuffer())
+                .then(buffer => WebAssembly.compile(buffer));
+            }
+        `;
+
+export default (new Transformer({
+  async transform({asset, options}) {
+    const asCompiler = await getASCompilerPath(asset.filePath, options);
+
+    const wasmFileName = `${path.basename(asset.filePath)}.wasm`;
+    const folder = tmpdir();
+    const wasmFilePath = path.join(folder, wasmFileName);
+    const sourceMapFileName = wasmFileName + '.map';
+    const sourceMapFilePath = path.join(folder, sourceMapFileName);
+    await asCompiler.ready;
+
+    await new Promise((resolve, reject) => {
+      const errorCollector = streamCollector();
+      const params = [
+        asset.filePath,
+        '-b',
+        wasmFilePath,
+        ...(options.sourceMapURLPattern
+          ? [
+              `--sourceMap=${renderNamePattern(options.sourceMapURLPattern, {
```

Comment:
Not sure how sourceMaps should be handled (renderNamePatter is a Rollup specific thing)